app/controller.py

```diff
@@ -66,8 +66,10 @@ def changeToSaveAs(self):
     self.host.changeFocusTo(self.host.interactiveSaveAs)
 
   def doCommand(self, ch):
+    #Check the commandSet for the input with both its string and integer representation.
     self.savedCh = ch
-    cmd = self.commandSet.get(ch)
+    keyName = app.curses_util.cursesKeyName(ch)
```

Comment:
I'm not sure what you mean by that, so tell me if this answers your question. 

* In the case that an integer is used, the corresponding key name (string) would be found and we can access the command that it corresponds to in the commandSet dictionary. 

* If a tuple is used, as it sometimes is (I've only encountered this for some ** ALT + CHAR** keys so far in some OS or terminals), nothing would happen because **cmd** in **doCommand()** would be **None** (as the tuple currently isn't in the dictionary), and so it will default to **defaultCommand()**. The default command for most of the controllers is **insertPrintable**. This function calls the function **curses.ascii.isprint(ch)** which returns whether a character is printable or not. In this case, **ALT + CHAR** is not printable so the function would exit and nothing would be printed. This behavior is also the same for if some integer keycode and it's string representation (if it has one) is not in the dictionary. For these **ALT + CHAR** shortcuts, I'm not sure how to go about them yet since they seem to be correct as integers for your system, but are tuples on my system. I'm currently working on figuring out a way to make the conversion work, so hopefully that goes well.

In summary, there should be no other bugs, other than the fact that some commands would not work if the keycodes or keynames are not the same for some particular system. I'd be good if you could confirm on your systems whether or not all the commands work